eureka/S4_generate_lightcurves/plots_s4.py

```diff
@@ -70,9 +70,60 @@ def drift1d(meta):
     else:
         plt.pause(0.2)
 
+def lc_driftcorr(meta, wave_1d, optspec):
+    '''Plot a 2D light curve with drift correction.
+
+    Parameters
+    ----------
+    meta:   MetaClass
+        The metadata object.
+    wave_1d:
+        Wavelength array with trimmed edges depending on xwindow and ywindow which have been set in the S3 ecf
+    optspec:
+        The optimally extracted spectrum.
+
+    Returns
+    -------
+    None
+    '''
+    plt.figure(4102, figsize=(8, 8))  # ev.n_files/20.+0.8))
+    plt.clf()
+    wmin = wave_1d.min()
+    wmax = wave_1d.max()
+    n_int, nx = optspec.shape
+    # iwmin       = np.where(ev.wave[src_ypos]>wmin)[0][0]
+    # iwmax       = np.where(ev.wave[src_ypos]>wmax)[0][0]
+    vmin = 0.97
+    vmax = 1.03
+    # normspec    = np.mean(ev.optspec,axis=1)/np.mean(ev.optspec[ev.inormspec[0]:ev.inormspec[1]],axis=(0,1))
+    normspec = optspec / np.mean(optspec, axis=0)
+    plt.imshow(normspec, origin='lower', aspect='auto', extent=[wmin, wmax, 0, n_int], vmin=vmin, vmax=vmax,
+               cmap=plt.cm.RdYlBu_r)
+    ediff = np.zeros(n_int)
+    for m in range(n_int):
+        ediff[m] = 1e6 * np.median(np.abs(np.ediff1d(normspec[m])))
+        # plt.scatter(ev.wave[src_ypos], np.zeros(nx)+m, c=normspec[m],
+        #             s=14,linewidths=0,vmin=vmin,vmax=vmax,marker='s',cmap=plt.cm.RdYlBu_r)
+    plt.title("MAD = " + str(np.round(np.mean(ediff), 0).astype(int)) + " ppm")
+    # plt.xlim(wmin,wmax)
+    # plt.ylim(0,n_int)
+    if meta.nspecchan > 1:
+        xticks = np.unique(np.concatenate([meta.wave_low,meta.wave_hi]))
```

Comment:
This doesn't look very good if you have many wavelength bins (ignore the fact that some of the wavelength bins are outside of the spectrum - I messed up the wavelength limits in my ecf). Perhaps you can angle the x-ticks by 45 degrees or so to help avoid them getting too crowded?

![fig4102-2D_LC](https://user-images.githubusercontent.com/24514122/156445944-41820a9b-e9b4-49f3-ab96-fd3efe626924.png)

Reply:
I picked 90 degrees to allow for maximum flexibility.  For my test case, there was no overlap in text with 40 channels.

Reply:
We could instead just change the dimensions of the figure rather than swapping the axes if we get to that point